cmd/api-mock/main.go

```diff
@@ -342,3 +366,29 @@ func unmarshalPayloadToAddonStatus(data []byte) (addonStatus, error) {
 	}
 	return status, nil
 }
+
+type APIMockConfig struct {
+	// Use string to simplify checking of values that weren't set
+	FailOnAddonStatusCreateEndpoint string `json:"failOnAddonStatusCreateEndpoint"`
```

Comment:
Can't `FailOnAddonStatusCreateEndpoint` be a bool instead based on the functionality it's serving?